parlai/crowdsourcing/tasks/turn_annotations_static/analysis/compile_results.py

```diff
@@ -0,0 +1,558 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import argparse
+import copy
+import json
+import os
+from datetime import datetime
+from typing import Any, Dict, Optional, Tuple
+
+import numpy as np
+import pandas as pd
+
+
+class TurnAnnotationsStaticResultsCompiler:
+    """
+    Class to compile results from static turn annotations.
+
+    Change PROBLEM_BUCKETS in task_config/annotation_buckets.json to be the buckets that
+    you are asking crowdsource workers to annotate with.
+    """
+
+    PROBLEM_BUCKETS = [
+        'bucket_0',
+        'bucket_1',
+        'bucket_2',
+        'bucket_3',
+        'bucket_4',
+        'none_all_good',
+    ]
+
+    NUM_SUBTASKS = 7
+    LIVE_ONBOARDING_IS_LAST_SUBTASK = True
+    LIVE_ONBOARDING_THRESHOLD = 0.5
+    INFLIGHT_ONBOARDING_DATA = None
+    NUM_ANNOTATIONS = 5
+
+    FILENAME_STUB = 'results'
+    CALCULATE_STATS_INTERANNOTATOR_AGREEMENT = True
+
+    @classmethod
+    def parse_args(cls):
+        parser = argparse.ArgumentParser()
+        parser.add_argument(
+            '--results-folders',
+            type=str,
+            help='Comma-separated list of result folders (example: "/basefolder/mephisto/data/runs/NO_PROJECT/123")',
+        )
+        parser.add_argument(
+            '--output-folder', type=str, help='Folder to save output files to'
+        )
+        parser.add_argument(
+            '--onboarding-in-flight-data-file',
+            type=str,
+            help='Path to JSONL file containing onboarding in-flight conversations',
+        )
+        parser.add_argument(
+            '--gold-annotations-file',
+            type=str,
+            default=None,
+            help='Path to a JSON file mapping utterance IDs to the gold annotations',
+        )
+        args = parser.parse_args()
+        return args
+
+    def __init__(self, opt: Optional[Dict[str, Any]] = None):
+        if opt is None:
+            opt = {}
+        if 'results_folders' in opt:
+            self.results_folders = opt['results_folders'].split(',')
+        else:
+            self.results_folders = None
+        self.output_folder = opt.get('output_folder')
+        self.onboarding_in_flight_data_file = opt.get('onboarding_in_flight_data_file')
+        self.gold_annotations_file = opt.get('gold_annotations_file')
+
+    def get_data_paths_mephisto(self, task_run_id_folder):
```

Comment:
why dont we use the databrowser? b/c read run id folder is faster?

Reply:
Good question - I'd consider direct folder access to be something of an anti-pattern, especially if this code intends to work with future Mephisto setups (internal may not use the MephistoDB that stores files locally, and the possible distributed MephsitoDB definitely wont have easy access like this).

Reply:
Good point, thanks! There are a lot of things left to refactor for these analysis scripts, so I've added this to the list of todos, and I've added a TODO on this line about that